docs/standard/linq/project-anonymous-type.md

```diff
@@ -0,0 +1,76 @@
+title: How to project an anonymous type - LINQ to XML
+description: You can project to an anonymous type rather than create a type just for use in the projection. This article provides an example for C# and Visual Basic.
+ms.date: 07/20/2015
+dev_langs:
+  - "csharp"
+  - "vb"
+ms.assetid: 5cb9be13-5ac4-4373-a034-b3520a5b2dec
+---
+# How to project an anonymous type (LINQ to XML)
+
+In some cases you might want to project a query to a new type, but the query would be your only use for the new type. Rather than create the type, you can project to an anonymous type. Anonymous types provide a convenient way to encapsulate a set of read-only properties in an object without having to explicitly define a type first. If you write a query that creates an object of an anonymous type in the `select` clause, the query returns an `IEnumerable` of the type.
+
+The following example shows shows creation of an object of an anonymous type that is initialized with two properties, `Amount` and `Message`.
+
+```csharp
+var v = new { Amount = 108, Message = "Hello" };
+```
+
+```vb
+Dim v = New With { .Amount = 108, .Message = "Hello" };
+```
+
+The type of each property is inferred by the compiler. The type name is generated by the compiler and isn't available at the source code level.
+
+For more information about anonymous types, see:
+
+- [Anonymous Types (C# Programming Guide)](../../csharp/programming-guide/classes-and-structs/anonymous-types)
+- [Anonymous Types (Visual Basic)](../../visual-basic/programming-guide/language-features/objects-and-classes/anonymous-types)
+
+## Example: Project an anonymous type by creating objects in the `select` clause
+
+In this example, the `select` clause projects an anonymous type. The example then uses `var` to create the `IEnumerable` object. Within the `foreach` loop, the iteration variable becomes an instance of the anonymous type created in the query expression.
+
+This example uses XML document [Sample XML file: Customers and orders](sample-xml-file-customers-orders.md).
+
+```csharp
+XElement custOrd = XElement.Load("CustomersOrders.xml");
+var custList =
+    from el in custOrd.Element("Customers").Elements("Customer")
+    select new {
+        CustomerID = (string)el.Attribute("CustomerID"),
+        CompanyName = (string)el.Element("CompanyName"),
+        ContactName = (string)el.Element("ContactName")
+    };
+foreach (var cust in custList)
+    Console.WriteLine("{0}:{1}:{2}", cust.CustomerID, cust.CompanyName, cust.ContactName);
+```
+
+```vb
+Dim custOrd As XElement = XElement.Load("CustomersOrders.xml")
+Dim custList = _
+    From el In custOrd.<Customers>.<Customer> _
+    Select New With { _
+        .CustomerID = el.@<CustomerID>, _
+        .CompanyName = el.<CompanyName>.Value, _
+        .ContactName = el.<ContactName>.Value _
+    }
+For Each cust In custList
+    Console.WriteLine("{0}:{1}:{2}", cust.CustomerID, cust.CompanyName, cust.ContactName)
+Next
+```
+
+The example produces this output:
```

Comment:
```suggestion
The example produces the following output:
```

Reply:
I standardized from the first on "The example produces this output", because I like to avoid "the following" (wordy). It is easy to change all the topics, so just let me know if you want me to do this.

Reply:
I actually prefer the following.

Reply:
My next PR will fix "the following" for all articles. Please just ignore it in the PRs I already created (the ones not yet reviewed and merged).